File: Dihya/assets/logos/serve_logos.js
// Serveur Node.js/Express ultra avancé pour servir les logos Dihya (sécurisé, RGPD, accessibilité, audit, multilingue)
const express = require('express');
const path = require('path');
const fs = require('fs');
const app = express();
const LOGOS_DIR = __dirname;

// Middleware audit/log RGPD
app.use((req, res, next) => {
  // Log anonymisé, RGPD
  console.log(`[LOGO_ACCESS] ${new Date().toISOString()} ${req.method} ${req.url}`);
  next();
});

// Endpoint pour servir un SVG
app.get('/logos/:logo', (req, res) => {
  const logoName = req.params.logo.replace(/[^a-zA-Z0-9\-_]/g, '');
  const logoPath = path.join(LOGOS_DIR, logoName + '.svg');
  if (fs.existsSync(logoPath)) {
    res.setHeader('Content-Type', 'image/svg+xml');
    res.setHeader('Access-Control-Allow-Origin', '*');
    res.sendFile(logoPath);
  } else {
    res.status(404).json({ error: 'Logo not found' });
  }
});

// Démarrage serveur
if (require.main === module) {
  const PORT = process.env.PORT || 4001;
  app.listen(PORT, () => {
    console.log(`Dihya Logos server running on port ${PORT}`);
  });
}

module.exports = app;

File: Dihya/assets/logos/serve_logos.py
"""
Serveur Python (Flask) ultra avancé pour servir les logos Dihya (sécurisé, RGPD, accessibilité, audit, multilingue)
"""
from flask import Flask, send_file, jsonify, request
import os

app = Flask(__name__)
LOGOS_DIR = os.path.dirname(os.path.abspath(__file__))

@app.before_request
def audit_log():
    print(f"[LOGO_ACCESS] {request.method} {request.path} {request.remote_addr}")

@app.route('/logos/<logo>')
def get_logo(logo):
    safe_logo = ''.join([c for c in logo if c.isalnum() or c in ('-', '_')])
    logo_path = os.path.join(LOGOS_DIR, safe_logo + '.svg')
    if os.path.exists(logo_path):
        return send_file(logo_path, mimetype='image/svg+xml')
    return jsonify({'error': 'Logo not found'}), 404

if __name__ == '__main__':
    app.run(host='0.0.0.0', port=4002)
